packages/runtime/runtime-utils/src/summarizerNode.ts

```diff
@@ -590,8 +578,8 @@ export class SummarizerNode implements ISummarizerNode {
     private trackingSequenceNumber: number;
     private constructor(
         private readonly logger: ITelemetryLogger,
-        private readonly summarizeInternalFn: (fullTree: boolean) => Promise<ISummarizeInternalResult>,
-        private readonly disabled: boolean,
+        private readonly summarizeInternalFn:
+            (fullTree: boolean, trackState: boolean) => Promise<ISummarizeInternalResult>,
```

Comment:
Having `trackState` in SummarizerNode seems like a little bit of the logic to circumvent SummarizerNode has leaked into the SummarizerNode itself. Probably not a big deal though, just a thought.

One way to avoid it is when handing off the lambda (i.e. `thisSummarizeInternal`) to create a SummarizerNode, you can hard-code `true` in the `trackState` parameter call.

Reply:
That's a great suggestion!